### ISSUE TITLE
does not start after new release! arch linux

### ISSUE THREAD

**Issue by wartemw** [ATLauncher/ATLauncher#363]
[25/7/2019 21:40:32 PM] Organising filesystem
[25/7/2019 21:40:32 PM] Error organising filesystem
[25/7/2019 21:40:32 PM] java.nio.file.FileAlreadyExistsException: /home/wartemw/Desktop/ATLaunther/instances
	at sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:429)
	at sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:262)
	at java.nio.file.Files.move(Files.java:1395)
	at com.atlauncher.FileSystem.renameDirectories(FileSystem.java:118)
	at com.atlauncher.FileSystem.organise(FileSystem.java:77)
	at com.atlauncher.App.main(App.java:267)

[25/7/2019 21:40:32 PM] ATLauncher Version: 3.3.0.0
[25/7/2019 21:40:32 PM] Operating System: Linux
[25/7/2019 21:40:32 PM] Java Version: Launcher: Java 8 (1.8.0_222), Minecraft: Java 8 (1.8.0_222)
[25/7/2019 21:40:32 PM] Java Path: /usr/lib/jvm/java-8-openjdk/jre
[25/7/2019 21:40:32 PM] 64 Bit Java: true
[25/7/2019 21:40:32 PM] RAM Available: 13963MB
[25/7/2019 21:40:32 PM] Launcher Directory: /home/wartemw/Desktop/ATLaunther
[25/7/2019 21:40:32 PM] Using Theme: Default Theme by RyanTheAllmighty
[25/7/2019 21:40:32 PM] Showing splash screen and loading everything
[25/7/2019 21:40:32 PM] Checking for updated files!
[25/7/2019 21:40:34 PM] Checking For Username Changes
[25/7/2019 21:40:34 PM] Checking For Username Changes Complete
[25/7/2019 21:40:34 PM] Launcher finished loading everything
[25/7/2019 21:40:34 PM] Launcher opening
[25/7/2019 21:40:34 PM] Made By Bob*
[25/7/2019 21:40:34 PM] *(Not Actually)
[25/7/2019 21:40:34 PM] Setting up Bottom Bar
[25/7/2019 21:40:34 PM] Finished Setting up Bottom Bar
[25/7/2019 21:40:34 PM] Setting up Tabs
[25/7/2019 21:40:35 PM] Finished Setting up Tabs
[25/7/2019 21:40:35 PM] Showing Launcher
[25/7/2019 21:40:48 PM] Logging into Minecraft!
[25/7/2019 21:40:49 PM] Added Account kenmancu
[25/7/2019 21:40:50 PM] Changed account to kenmancu
[25/7/2019 21:40:50 PM] Downloading skin for kenmancu
[25/7/2019 21:40:51 PM] Changed account to kenmancu
[25/7/2019 21:41:08 PM] Started install of TechNodefirmacraft version 3.7.1
[25/7/2019 21:41:39 PM] java.lang.NullPointerException
	at com.atlauncher.workers.InstanceInstaller.lambda$downloadLibraries$15(InstanceInstaller.java:774)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:485)
	at com.atlauncher.workers.InstanceInstaller.downloadLibraries(InstanceInstaller.java:771)
	at com.atlauncher.workers.InstanceInstaller.install(InstanceInstaller.java:335)
	at com.atlauncher.workers.InstanceInstaller.doInBackground(InstanceInstaller.java:178)
	at com.atlauncher.workers.InstanceInstaller.doInBackground(InstanceInstaller.java:77)
	at javax.swing.SwingWorker$1.call(SwingWorker.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at javax.swing.SwingWorker.run(SwingWorker.java:334)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

[25/7/2019 21:41:52 PM] Copied Log to clipboard

**Comment by jonahgraham** [comment]
I am seeing almost the same stack trace (see below). More info to reproduce:

- I am on Ubuntu 18.04 with openjdk version "1.8.0_212"
- I cannot reproduce the problem on Windows 10
- The exception happens when creating a Vanilla Minecraft instance with version 1.8.9 but not with newer versions
- I tried in a clean/new directory to make sure nothing old was causing the problem

```
[25/7/2019 14:57:41 PM] java.lang.NullPointerException
	at com.atlauncher.workers.InstanceInstaller.lambda$downloadLibraries$15(InstanceInstaller.java:774)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at com.atlauncher.workers.InstanceInstaller.downloadLibraries(InstanceInstaller.java:771)
	at com.atlauncher.workers.InstanceInstaller.install(InstanceInstaller.java:335)
	at com.atlauncher.workers.InstanceInstaller.doInBackground(InstanceInstaller.java:178)
	at com.atlauncher.workers.InstanceInstaller.doInBackground(InstanceInstaller.java:77)
	at javax.swing.SwingWorker$1.call(SwingWorker.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at javax.swing.SwingWorker.run(SwingWorker.java:334)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by RyanTheAllmighty** [comment]
Sorry guys, I do majority of my testing in Windows, and only some small exploratory testing in linux and osx.

This has been fixed in master and will go out with 3.3.0.1 of the launcher

**Comment by RyanTheAllmighty** [comment]
Released with version 3.3.0.1

**Comment by jonahgraham** [comment]
Thank you @RyanTheAllmighty - my use case is now working with 3.3.0.1.
